src/cpp/session/SessionSourceDatabase.cpp

```diff
@@ -533,9 +531,9 @@ void SourceDocument::writeToJson(json::Object* pDocJson, bool includeContents) c
 {
    json::Object& jsonDoc = *pDocJson;
    jsonDoc["id"] = id();
-   jsonDoc["path"] = !path().empty() ? path_ : json::Value();
+   jsonDoc["path"] = !path().empty() ? json::Value(path_) : json::Value();
    jsonDoc["project_path"] = pathToProjectPath(path_);
-   jsonDoc["type"] = !type().empty() ? type_ : json::Value();
+   jsonDoc["type"] = !type().empty() ? json::Value(type_) : json::Value();
```

Comment:
Is it now a requirement to wrap into a `json::Value` when assigning via regular c++ object?

Reply:
I did make the constructors explicit, but actually assigning should still work. Let me look into this.

Reply:
Ah, so if we were assigning this directly (e.g. `jsonDoc["type"] = type_`), assignment operators would work and an explicit conversion wouldn't be necessary. Because we are effectively returning `type_` from the tertiary operator, it wants to call the conversion constructor which is now explicit.